### ISSUE TITLE
Shared doc doesn't appear on initial page load for restricted member

### ISSUE THREAD

**Issue by ctrance** [mobilizingcs/document_tool#7]
To reproduce this bug, please follow the steps below:
1. Clear the history
2. Log in as ctrance2 or ctrance4
3. Doc list didn't show shared file Jellyfish.jpg
4. Refresh the page, only then the shared file will appear.

**Comment by stevenolen** [comment]
I attempted to reproduce this as follows and was unable to:

1. added mobilize-teacher

**Comment by stevenolen** [comment]
I attempted to reproduce with the following (sorry for the rogue comment):
1. added mobilize-teacher to cathy:test class
2. logged in as mobilize teacher from incognito window and private browsing window in FF  (which gives me a history and cache-free browsing experience).
3. I was able to view Jellyfish and another shared doc that mobilize-teacher only has a restricted member permission for on first load.

I'm not really sure what could be going on in this case, but I don't use any permanent storage in this app and do a `/app/document/read` call on every page refresh, getting all data from the server each time. The server would have to return two different sets of data each time in order for this to happen (which could happen if the state changed between refreshes).

@ctrance if this is still happening, could I have you re-try and provide me some info from the console? I added a logger to print out a big json blob of the result from the server, so we can compare the output from the server on each page load:
1. load page on first try, and if doc is not available, provideconsole console content
2. load page on second try, if doc has appeared, provide console again.
   (feel free to email these two blobs to me, since they may content semi-private data).
